proxy/src/schema.rs

```diff
@@ -223,36 +344,28 @@ impl Query {
         "1.0"
     }
 
-    fn blob(ctx: &Context, id: IdInput, revision: String, path: String) -> FieldResult<Blob> {
-        let repo = GitRepository::new(&ctx.dummy_repo_path).expect("setting up repo failed");
-        let mut browser = GitBrowser::new(&repo).expect("setting up browser for repo failed");
+    fn blob(ctx: &Context, id: IdInput, revision: String, path: String) -> Result<Blob, Error> {
+        let repo = GitRepository::new(&ctx.dummy_repo_path)?;
+        let mut browser = GitBrowser::new(&repo)?;
 
         // Best effort to guess the revision.
-        if let Err(err) = browser
+        browser
             .branch(BranchName::new(&revision))
             .or(browser.commit(Sha1::new(&revision)))
-            .or(browser.tag(TagName::new(&revision)))
-        {
-            let err_fmt = format!("{:?}", err);
-
-            return Err(FieldError::new(
-                "Git error occurred",
-                graphql_value!({ "git": err_fmt }),
-            ));
-        };
+            .or(browser.tag(TagName::new(&revision)))?;
 
-        let root = browser
-            .get_directory()
-            .expect("unable to get root directory");
+        let root = browser.get_directory()?;
 
         let mut p = Path::root();
         p.append(&mut Path::from_string(&path));
-        let file = root
-            .find_file(&p)
-            .expect(&format!("unable to find file: {} -> {}", path, p));
-        let last_commit = browser
-            .last_commit(&p)
-            .expect(&format!("[blob] unable to get last commit: {}", p));
+        let file = match root.find_file(&p) {
+            Some(result) => Ok(result),
+            None => Err(Error::FileNotFound(format!("unable to find file: {} -> {}", path, p))),
```

Comment:
And this pattern is captured with `ok_or`, where it turns `Some` into `Ok` and `None` into an `Err` with the error you supply